lib/upipe-freetype/upipe_freetype.c

```diff
@@ -472,6 +697,46 @@ static int upipe_freetype_control_real(struct upipe *upipe,
             return upipe_freetype_set_option(upipe, option, value);
         }
 
+        case UPIPE_FREETYPE_GET_BBOX: {
+            if (ubase_get_signature(args) != UPIPE_FREETYPE_SIGNATURE)
+                return UBASE_ERR_UNHANDLED;
+
+            UBASE_SIGNATURE_CHECK(args, UPIPE_FREETYPE_SIGNATURE);
+            const char *str = va_arg(args, const char *);
+            struct upipe_freetype_bbox *bbox_p =
+                va_arg(args, struct upipe_freetype_bbox *);
+            return upipe_freetype_get_bbox_real(upipe, str, bbox_p);
+        }
+
+        case UPIPE_FREETYPE_GET_WIDTH: {
+            if (ubase_get_signature(args) != UPIPE_FREETYPE_SIGNATURE)
+                return UBASE_ERR_UNHANDLED;
+
+            UBASE_SIGNATURE_CHECK(args, UPIPE_FREETYPE_SIGNATURE);
+            const char *str = va_arg(args, const char *);
+            uint64_t *width_p = va_arg(args, uint64_t *);
+            return upipe_freetype_get_width_real(upipe, str, width_p);
+        }
+
+        case UPIPE_FREETYPE_SET_PIXEL_SIZE: {
+            if (ubase_get_signature(args) != UPIPE_FREETYPE_SIGNATURE)
+                return UBASE_ERR_UNHANDLED;
+
+            UBASE_SIGNATURE_CHECK(args, UPIPE_FREETYPE_SIGNATURE);
+            unsigned pixel_size = va_arg(args, unsigned);
+            return upipe_freetype_set_pixel_size_real(upipe, pixel_size);
+        }
+
+        case UPIPE_FREETYPE_SET_BASELINE: {
+            if (ubase_get_signature(args) != UPIPE_FREETYPE_SIGNATURE)
+                return UBASE_ERR_UNHANDLED;
+
+            UBASE_SIGNATURE_CHECK(args, UPIPE_FREETYPE_SIGNATURE);
+            uint64_t xoff = va_arg(args, uint64_t);
+            uint64_t yoff = va_arg(args, uint64_t);
+            return upipe_freetype_set_baseline_real(upipe, xoff, yoff);
+        }
+
```

Comment:
What about splitting the switch and factoring out the signature checks?

Reply:
Done